contracts/Poll.sol

```diff
@@ -0,0 +1,232 @@
+    //     // assigns reference
+    //     Voter storage sender = voters[msg.sender];
+    //     require(!sender.voted, "You already voted.");
+
+    //     require(to != msg.sender, "Self-delegation is disallowed.");
+
+    //     // Forward the delegation as long as
+    //     // `to` also delegated.
+    //     // In general, such loops are very dangerous,
+    //     // because if they run too long, they might
+    //     // need more gas than is available in a block.
+    //     // In this case, the delegation will not be executed,
+    //     // but in other situations, such loops might
+    //     // cause a contract to get "stuck" completely.
+    //     while (voters[to].delegate != address(0)) {
+    //         to = voters[to].delegate;
+
+    //         // We found a loop in the delegation, not allowed.
+    //         require(to != msg.sender, "Found loop in delegation.");
+    //     }
+
+    //     // Since `sender` is a reference, this
+    //     // modifies `voters[msg.sender].voted`
+    //     Voter storage delegate_ = voters[to];
+
+    //     // Voters cannot delegate to wallets that cannot vote.
+    //     require(delegate_.weight >= 1);
+    //     sender.voted = true;
+    //     sender.delegate = to;
+    //     if (delegate_.voted) {
+    //         // If the delegate already voted,
+    //         // directly add to the number of votes
+    //         proposals[delegate_.vote].voteCount += sender.weight;
+    //     } else {
+    //         // If the delegate did not vote yet,
+    //         // add to her weight.
+    //         delegate_.weight += sender.weight;
+    //     }
+    // }
+
+    // function cancel delegation
+
+
+    modifier newPollCreateionCheck(string memory name, string memory desc, uint dur, Selection[] sel) 
+    {
+        require(bytes(name).length > 0, "Poll name is empty");
+        require(bytes(desc).length > 0, "Poll description is empty");
+        require(dur > 0, "Poll duration is empty");
+        require(bytes(sel).length > 0, "Choice to select from not given");
+        
+        _;
+    }
+
+    function createPoll(string memory name, string memory desc, uint dur, bool blind, bool aboutDAO, Selection[] sel) 
+        newPollCreateionCheck(name, desc, dur, sel) public
+    {
+
+        require(bytes(participants[msg.sender]).length > 0, "Not registered in the system");
+
+        PollEvent memory newPoll = PollEvent(State.VOTING, nextPollId, name, desc, dur, blind, aboutDAO, sel);
+        polls[nextPollId] = newPoll;
+
+        emit pollCreated(msg.sender, name, dur, blind, aboutDAO);
+    }
+
+    modifier voteCheck(uint pollId, Selection choice) 
+    {
+        PollEvent storage thisPoll = polls[pollId];
+        require(bytes(thisPoll).length > 0, "Poll not created");
+        require(thisPoll.state == State.VOTING, "The poll has ended");
+        require(bytes(choice).length > 0, "No choice made yet");
+
+        _;
+    }
+
+    function vote(uint pollId, Selection choice)
```

Comment:
I feel like we should have a map mapping between participants and their voting status: 
for example, [participant # 1: Jerry; False]: Jerry hasn't voted yet. 
Is that what you're trying to do here?

Reply:
If we use the way you suggested, we can only record Jerry's vote for one poll, but we want to record down his vote for every poll, that's why I store the voting information in ` address[] voted // An array to store who has already voted` and  `Selection[] votedChoices; // An array to store voter's choices `